store/driver/txn/ranged_kv_retriever.go

```diff
@@ -62,3 +65,145 @@ func (s *RangedKVRetriever) Intersect(startKey, endKey kv.Key) *RangedKVRetrieve
 
 	return nil
 }
+
+// ScanCurrentRange scans the retriever for current range
+func (s *RangedKVRetriever) ScanCurrentRange(reverse bool) (kv.Iterator, error) {
+	if !s.Valid() {
+		return nil, errors.New("retriever is invalid")
+	}
+
+	startKey, endKey := s.StartKey, s.EndKey
+	if !reverse {
+		return s.Iter(startKey, endKey)
+	}
+
+	iter, err := s.IterReverse(endKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(startKey) > 0 {
+		iter = newLowerBoundReverseIter(iter, startKey)
+	}
+
+	return iter, nil
+}
+
+type sortedRetrievers []*RangedKVRetriever
+
+func (retrievers sortedRetrievers) TryGet(ctx context.Context, k kv.Key) (bool, []byte, error) {
+	if len(retrievers) == 0 {
+		return false, nil, nil
+	}
+
+	_, r := retrievers.searchRetrieverByKey(k)
+	if r == nil || !r.Contains(k) {
+		return false, nil, nil
+	}
+
+	val, err := r.Get(ctx, k)
+	return true, val, err
+}
+
+func (retrievers sortedRetrievers) TryBatchGet(ctx context.Context, keys []kv.Key, collectF func(k kv.Key, v []byte)) ([]kv.Key, error) {
+	if len(retrievers) == 0 {
+		return keys, nil
+	}
+
+	nonCustomKeys := make([]kv.Key, 0, len(keys))
```

Comment:
it can be waste to allocate by length of `keys`. I prefer `var nonCustomKeys []kv.Key`.

Reply:
I think in most cases, nonCustomKeys has a length of input keys (reading non-temporary table). So make its capacity from len(keys) to 0 can not make a better performance. But I can provide another way to alloc the array on demand.

Reply:
Done